sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/implementation/ReactorSessionCacheTest.java

```diff
@@ -0,0 +1,273 @@
+
+        // Since the session0a was evicted, the cache should return the new session, session0b.
+        StepVerifier.create(cache.getOrLoad(Mono.just(connection), session0Name, cacheLoader))
+            .expectNext(session0b)
+            .verifyComplete();
+    }
+
+    @Test
+    void shouldNotEvictSessionIfOwnerDisposed() {
+        final ReactorSessionCache cache = createSessionCache();
+        final Connection connection = mock(Connection.class);
+
+        final String session0Name = "session-0";
+        final Sinks.Many<AmqpEndpointState> session0EndpointStates = endpointStatesSink();
+        final ReactorSession session0 = session(session0Name, endpointStatesSink(), null);
+
+        final HashMap<String, Deque<ReactorSession>> cacheLoaderLookup = new HashMap<>();
+        cacheLoaderLookup.put(session0Name, sessions(session0));
+        final CacheLoader cacheLoader = new CacheLoader(cacheLoaderLookup);
+
+        StepVerifier.create(cache.getOrLoad(Mono.just(connection), session0Name, cacheLoader))
+            .expectNext(session0)
+            .verifyComplete();
+
+        // Signal that the cache owner (Connection) is disposed.
+        cache.setOwnerDisposed();
+        // Signal that session0a is terminated after owner disposal.
+        session0EndpointStates.emitComplete(Sinks.EmitFailureHandler.FAIL_FAST);
+
+        // Since the owner is disposed, the cache should not evict the session0 even if it's terminated.
+        StepVerifier.create(cache.getOrLoad(Mono.just(connection), session0Name, cacheLoader))
+            .expectNext(session0)
+            .verifyComplete();
+    }
+
+    private ReactorSessionCache createSessionCache() {
+        return new ReactorSessionCache(CONNECTION_ID, NAMESPACE, handlerProvider, reactorProvider, OPEN_TIMEOUT,
+            logger);
+    }
+
+    private Sinks.Many<AmqpEndpointState> endpointStatesSink() {
+        return Sinks.many().replay().latestOrDefault(AmqpEndpointState.UNINITIALIZED);
+    }
+
+    private static Deque<ReactorSession> sessions(ReactorSession... sessions) {
+        final Deque<ReactorSession> queue = new ArrayDeque<>(sessions.length);
+        Collections.addAll(queue, sessions);
+        return queue;
+    }
+
+    private static ReactorSession session(String sessionName, Sinks.Many<AmqpEndpointState> sink, Throwable openError) {
+        final ReactorSession session = mock(ReactorSession.class);
+        when(session.getSessionName()).thenReturn(sessionName);
+        when(session.getEndpointStates()).thenReturn(sink.asFlux());
+        if (openError != null) {
+            when(session.open()).thenReturn(Mono.error(openError));
+        } else {
+            when(session.open()).thenReturn(Mono.just(session));
+        }
+        when(session.closeAsync(anyString(), any(), eq(true))).thenReturn(Mono.empty());
+        return session;
+    }
+
+    private static final class CacheLoader implements ReactorSessionCache.Loader {
```

Comment:
Why not `mock(Loader.class)` since it is an interface?

Reply:
That also works, this was just a preference as I was trying to limit mock usage just to above _session(String, ..)_